forge/test/helpers/PoolSetupHelper.t.sol

```diff
@@ -0,0 +1,144 @@
+        oracleWrapper = new ChainlinkOracleWrapper(address(chainlinkOracle), DEPLOYER);
+
+        // keeper oracle
+        settlementEthOracle = new ChainlinkOracleWrapper(address(ethOracle), DEPLOYER);
+
+        // Deploy and init pool
+        poolFactory = new PoolFactory(FEE_RECEIVER, GOVERNANCE);
+
+        invariantCheck = new InvariantCheck(address(poolFactory));
+        poolFactory.setInvariantCheck(address(invariantCheck));
+
+        poolKeeper = new PoolKeeper(address(poolFactory));
+        poolFactory.setPoolKeeper(address(poolKeeper));
+        poolFactory.setFee(DEFAULT_FEE);
+
+        keeperRewards = new KeeperRewards(address(poolKeeper));
+        poolKeeper.setKeeperRewards(address(keeperRewards));
+
+        autoclaim = new AutoClaim(address(poolFactory));
+        poolFactory.setAutoClaim(address(autoclaim));
+    }
+
+    /**
+     * Deploys a new instance of a pool, as well as an ERC20 token to use as a settlement token.
+     * @param poolCode The pool identifier
+     * @param frontRunningInterval The front running interval value. Must be less than the update interval
+     * @param updateInterval The update interval value
+     * @param leverage The amount of leverage the pool will apply
+     * @param feeAddress The address to transfer fees to on a fund movement
+     * @param fee The fund movement fee.
+     * @param mintFee Mint fee
+     * @param burnFee Burn fee
+     * @param changeInterval Change interval
+     */
+    function setupPoolAndTokenContracts(
+        string memory poolCode,
+        uint32 frontRunningInterval,
+        uint32 updateInterval,
+        uint16 leverage,
+        address feeAddress,
+        uint256 fee,
+        uint256 mintFee,
+        uint256 burnFee,
+        uint256 changeInterval
+    ) internal {
+        skip(100000); // start block.timestamp @ 100000
+        vm.startPrank(GOVERNANCE);
+        _deployPoolSetupContracts();
+
+        if (fee > 0) {
+            poolFactory.setFee(fee);
+        }
+
+        if (feeAddress != address(0)) {
+            poolFactory.setFeeReceiver(feeAddress);
+        }
+
+        vm.stopPrank();
+
+        // deploy leveraged pool
+        IPoolFactory.PoolDeployment memory deploymentParameters = IPoolFactory.PoolDeployment({
+            poolName: poolCode,
+            frontRunningInterval: frontRunningInterval,
+            updateInterval: updateInterval,
+            leverageAmount: leverage,
+            settlementToken: address(token),
+            oracleWrapper: address(oracleWrapper),
+            settlementEthOracle: address(settlementEthOracle),
+            feeController: DEPLOYER,
+            mintingFee: mintFee,
+            burningFee: burnFee,
+            changeInterval: changeInterval
+        });
+
+        vm.prank(DEPLOYER);
+        leveragedPool = LeveragedPool(poolFactory.deployPool(deploymentParameters));
+
+        // fast forward in seconds
+        skip(updateInterval * 10_000);
```

Comment:
Why multiple by 10,000? `updateInterval` is already in seconds.

Reply:
That comment is informational - I would imagine that's an arbitrary number so that some time passes before performing pool upkeep? Equivalent to this line https://github.com/tracer-protocol/perpetual-pools-contracts/blob/pools-v2/test/utilities.ts#L302

Reply:
Hmmm I'm not 100% about that line in the tests, but it's multiplying by 1k to get it to milliseconds, then another 1k just because.

In our case here, it will be 10000 update intervals, which will require a number of separate upkeeps before it "catches up".

You could get away with just `skip(updateInterval + 20)` or some smaller value, to make sure it's able to upkeep everything straight away.